supertokens_python/recipe/openid/utils.py

```diff
@@ -85,4 +94,24 @@ def validate_and_normalise_user_input(
         OverrideConfig(functions=override.functions, apis=override.apis),
         issuer_domain,
         issuer_path,
+        is_issuer_domain_given,
+    )
+
+
+async def get_issuer_domain_or_throw_error(
+    issuer_domain: Optional[str],
+    config: OpenIdConfig,
+    app_info: AppInfo,
+    user_context: Any,
+) -> str:
+    if issuer_domain is not None:
+        return issuer_domain
+    if config.is_issuer_domain_given:
+        issuer_domain_resp = await config.issuer_domain(None, user_context)  # type: ignore
+        return issuer_domain_resp.get_as_string_dangerous()
+    if app_info.initial_api_domain_type == "string":
+        issuer_domain_resp = await config.issuer_domain(None, user_context)  # type: ignore
+        return issuer_domain_resp.get_as_string_dangerous()
+    raise Exception(
```

Comment:
Create an issue or add to the TODOs in the base PR for all the pending TODOs so far. And also add that all the error copies need to be done better